adoc/chapters/architecture.adoc

```diff
@@ -1099,23 +1098,33 @@ to the buffer or USM allocation from device code.
 
 ==== Forward progress
 
-Memory consistency guarantees are independent of any forward progress
-guarantees.  A SYCL implementation must execute work-items concurrently
-and must ensure that the work-items in a group obey the semantics of
-<<group-barrier,group barriers>>, but are not required to provide any
-additional forward progress guarantees.
+This section, and any subsequent section referring to progress guarantees, uses
+the following terms as defined in the {cpp}17 specification: thread of
```

Comment:
```suggestion
the following terms as defined in the {cpp} core language: thread of
```

We state in section 3.3 "Normative references" that the SYCL specification refers to C++17 as the "C++ core language".  Doing it this way avoid the need to update lots of references in our spec when we move to a newer version of C++.

Reply:
Thanks, Greg.  Fixed!